### PR TITLE
[Coroutines][Docs] Add a discussion on the handling of certain parameter attribs

### PR DESCRIPTION
**Pull request by TylerNowicki**
ByVal arguments and Swifterror require special handling in the coroutine passes. The goal of this section is to provide a description of how these parameter attributes are handled.

### CODE REVIEW

**Comment by TylerNowicki** [comment]
Ping, any comments on the parameter attributes section?

File: llvm/docs/Coroutines.rst [hunk 1]
```diff
@@ -810,6 +810,28 @@ The LLVM IR for a coroutine using a Coroutine with a custom ABI looks like:
     ret ptr %hdl
   }
 
+Parameter Attributes
+====================
+Some parameter attributes, used to communicate additional information about the result or parameters of a function, require special handling.
+
+ByVal
+-----
+A ByVal parameter on an argument indicates that the pointer parameter should really be passed by value to the function.
```
Comment (ChuanqiXu9) [inline llvm/docs/Coroutines.rst:819]: `the pointer parameter` maybe the pointee?

**Review by ChuanqiXu9 — COMMENTED** [review]
Looks not bad to me. But given it relates to Swift, I'd like to leave this to @rjmccall to give the formal approve.

File: llvm/docs/Coroutines.rst (hunk 1 above)
Comment (TylerNowicki) [inline llvm/docs/Coroutines.rst:819]:
```suggestion
A ByVal parameter on an argument indicates that the pointee should be treated as being passed by value to the function.
```

**Comment by TylerNowicki** [comment]
@rjmccall Ping, this is documentation for how the parameter attribs ByVal and SwiftError should be handled by the coroutine passes.

**Review by rjmccall — APPROVED** [review]
Yes, this looks accurate.

**Pull request merged**